internal/network/socket_unix.go

```diff
@@ -0,0 +1,24 @@
+package network
+
+import (
+	"fmt"
+	"os"
+	"runtime/debug"
+
+	"golang.org/x/sys/unix"
+)
+
+func (s *socketFD) release(fd int) {
+	s.releaseFunc(fd, closeTraceError)
+}
+
+func (s *socketFD) close() {
+	s.closeFunc(closeTraceError)
+}
+
+func closeTraceError(fd int) {
+	if err := unix.Close(fd); err != nil {
+		fmt.Fprintf(os.Stderr, "close(%d) => %s\n", fd, err)
+		debug.PrintStack()
+	}
+}
```

Comment:
Leftover from debugging?

Reply:
Yes but I want to keep it because bugs caused by race conditions that result in bad resource management is really difficult to track down, we'll be happy if we ever get a stack trace here!